app/models/document.rb

```diff
@@ -184,14 +184,22 @@ def self.all(page, per_page, q: nil)
   def self.find(content_id)
     response = Services.publishing_api.get_content(content_id)
 
-    if response
-      self.from_publishing_api(response.to_hash)
+    raise RecordNotFound, "Document: #{content_id}" unless response
+
+    attributes = response.to_hash
+    document_type = attributes.fetch("document_type")
+    document_class = document_type.camelize.constantize
+
+    if [document_class, Document].include?(self)
+      document_class.from_publishing_api(response.to_hash)
     else
-      raise RecordNotFound
+      message = "#{self}.find('#{content_id}') returned the wrong type: '#{document_class}'"
+      raise TypeMismatchError, message
     end
```

Comment:
I think `descendants` might be incorrect if Rails autoloading hasn't loaded all of the models yet. I'm going to merge as-is and we can change it later if needed.